src/language/dafnyInstallation.ts

```diff
@@ -89,15 +89,41 @@ export async function getCompilerRuntimePath(context: ExtensionContext): Promise
   return configuredPath;
 }
 
+// We cache the language server runtime path so that we don't need to copy it every time.
+let LanguageServerRuntimePath: string | null = null;
+
 export async function getLanguageServerRuntimePath(context: ExtensionContext): Promise<string> {
-  const configuredPath = await ifNullOrEmpty(
-    getConfiguredLanguageServerRuntimePath(),
+  if(LanguageServerRuntimePath != null) {
+    return LanguageServerRuntimePath;
+  }
+  const configuredLanguageServerPath = getConfiguredLanguageServerRuntimePath();
+  let mightBeRecomputed = false;
+  if(configuredLanguageServerPath !== '' && configuredLanguageServerPath != null) {
+    mightBeRecomputed = true;
+  }
+
+  let configuredPath = await ifNullOrEmpty(
+    configuredLanguageServerPath,
     async () => LanguageServerConstants.GetDefaultPath(await getConfiguredVersion(context))
   );
-  if(path.isAbsolute(configuredPath)) {
-    return configuredPath;
+  if(!path.isAbsolute(configuredPath)) {
+    configuredPath = path.join(context.extensionPath, configuredPath);
   }
-  return path.join(context.extensionPath, configuredPath);
+  if(mightBeRecomputed && (/\.dll$/.exec(configuredPath)) != null) {
+    // We copy DafnyLanguageServer.dll to another location
+    // so that rebuilding Dafny will not fail because it's open in VSCode
+    const newPath = configuredPath.replace('.dll', '-vscode.dll');
```

Comment:
Careful, this will replace only the first occurence of `.dll` in the string, not the last. So if there are multiple `.dll` bits within the string, you'll end up with something different.
In this case, you can go back to your regex approach to make sure it only replaces the endling `.dll`, Something like this:

```suggestion
    const newPath = configuredPath.replace(/\.dll$/, '-vscode.dll');
```